src/puppeteer-provider.ts

```diff
@@ -300,6 +300,10 @@ export class PuppeteerProvider {
 
     jobdebug(`${jobId}: ${req.url}: Inbound WebSocket request.`);
 
+    socket.on('error', (err: Error) => {
+      jobdebug(`${jobId}: A socket error has occurred: ${err.stack}`);
```

Comment:
I'd love to follow up with you on this and see what the errors are. It might be best to close the session if the box is not able to handle the session (and we can error out as well).

Reply:
This is the error we've observed when the error was not handled: 

```2020-01-24T09:16:42.021Z browserless:job V45N0C04JWNM5ERD2ULGNE3YUEIGQUWI: /: Inbound WebSocket request.
2020-01-24T09:16:42.022Z browserless:job V45N0C04JWNM5ERD2ULGNE3YUEIGQUWI: Adding new job to queue.
2020-01-24T09:16:42.022Z browserless:server undefined: Recording queued stat.
2020-01-24T09:16:42.022Z browserless:job 6DEBFIQXBL5ESGKNWFQOMFV340WU5DBG: /: Inbound WebSocket request.
2020-01-24T09:16:42.022Z browserless:job 6DEBFIQXBL5ESGKNWFQOMFV340WU5DBG: Adding new job to queue.
2020-01-24T09:16:42.022Z browserless:server undefined: Recording queued stat.
2020-01-24T09:16:42.022Z browserless:job R7AEJNMSZV79XEV605MBK93HKQA4GAXF: /: Inbound WebSocket request.
2020-01-24T09:16:42.022Z browserless:job R7AEJNMSZV79XEV605MBK93HKQA4GAXF: Adding new job to queue.
2020-01-24T09:16:42.022Z browserless:server undefined: Recording queued stat.
2020-01-24T09:16:42.022Z browserless:job 5KV6CG3GRL600MMJ6GQ1QMO6G1B6ZFAH: /: Inbound WebSocket request.
2020-01-24T09:16:42.022Z browserless:job 5KV6CG3GRL600MMJ6GQ1QMO6G1B6ZFAH: Too many concurrent and queued requests, rejecting with 429.
2020-01-24T09:16:42.022Z browserless:server /: Too Many Requests
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:201:27)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: 'ECONNRESET',
  code: 'ECONNRESET',
  syscall: 'read'
}```

Basically just that the TCP connection was dropped, but why it happened, I have no idea. As I mentioned, this only happened when both Browserless and our puppeteer client was running on our Kubernetes cluster. When I ran them both on Amazon ECS or on my local machine, there were no problems. In fact, it was enough to run the client on my local machine and Browserless on Kubernetes and the problem went away.

Unfortunately this is all I know, but please tell me if I can check anything else